acceptance/setup/common/pre-suite/031_add_local_nix_user.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require 'bolt_setup_helper'
+
+test_name "Configure a bolt local_user on non-windows" do
+  extend Beaker::HostPrebuiltSteps
+  extend Acceptance::BoltSetupHelper
+
+  skip_test('Not applicable windows') if bolt['platform'] =~ /windows/
+
+  # Use Bolt's puppet for package testing
+  result = on(bolt, 'stat /usr/local/bin/puppet', acceptable_exit_codes: [0, 1])
+  puppet_path = result.exit_code == 0 ? '/usr/local/bin/puppet' : '/opt/puppetlabs/bolt/bin/puppet'
+
+  case bolt['platform']
+  when /osx/
```

Comment:
Down to just two issues for this: 
1. This manifest does not create a new user for osx 10.14. Error like the following: 
```
[1] pry(#<Beaker::TestCase>)> r = on(bolt, "#{puppet_path} apply #{dir}/osx_user.pp")
=> #<Beaker::Result:0x000056259ad72330
 @cmd=
  "/usr/local/bin/puppet apply /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/osx_manifest.XXXXXX.hrS7odwF/osx_user.pp",
 @exit_code=0,
 @host="osx1014-64-1",
 @output=
  "\e[mNotice: Compiled catalog for ytn2i49zcmp4zgr.delivery.puppetlabs.net in environment production in 0.14 seconds\e[0m\n\e[1;31mError: Could not set the dscl NFSHomeDirectory key with value: /Users/local_user - #<Puppet::ExecutionFailure: Execution of '/usr/bin/dscl . -merge /Users/local_user NFSHomeDirectory /Users/local_user' returned 40: <main> attribute status: eDSPermissionError\n<dscl_cmd> DS Error: -14120 (eDSPermissionError)>\e[0m\n\e[1;31mError: /Stage[main]/Main/User[local_user]/ensure: change from 'absent' to 'present' failed: Could not set the dscl NFSHomeDirectory key with value: /Users/local_user - #<Puppet::ExecutionFailure: Execution of '/usr/bin/dscl . -merge /Users/local_user NFSHomeDirectory /Users/local_user' returned 40: <main> attribute status: eDSPermissionError\n<dscl_cmd> DS Error: -14120 (eDSPermissionError)>\e[0m\n\e[mNotice: /Stage[main]/Main/Exec[/usr/sbin/createhomedir -c -l -u local_user]/returns: executed successfully\e[0m\n\e[mNotice: Applied catalog in 1.57 seconds\e[0m\n",

```
2. Cannot sudo on redhat 7 without tty
```
15:32:04 Beaker::Host::CommandFailure: Host 'j07wsvmhylkb1st.delivery.puppetlabs.net' exited with 2 running:
15:32:04  bolt script run /home/local_user/test_local.sh hello --nodes localhost --run-as 'local_user'
15:32:04 Last 10 lines of output were:
15:32:04 	Started on localhost...
15:32:04 	Failed on localhost:
15:32:04 	  Could not make tempdir: sudo: sorry, you must have a tty to run sudo
15:32:04 	Failed on 1 node: localhost
15:32:04 	Ran on 1 node in 0.01 seconds
```
Which perhaps we could solve with something like: https://github.com/puppetlabs/bolt/blob/44c4bde0db1b532df82239f2dc5d956076958a3c/acceptance/tests/apply_plan_ssh.rb#L134-L143

Reply:
Sweet. Did some digging in beaker and found https://github.com/puppetlabs/beaker/blob/6fa2744a5dec68b694657aab86f4884e7e21bd6b/lib/beaker/host/mac/user.rb#L47 which appears to solve my osx issue.